create_blog_post_using_azure_openai_and_publish_in_medium/main.bal

```diff
@@ -0,0 +1,49 @@
+import ballerina/http;
+import ballerina/io;
+import ballerinax/azure_openai.text;
+import ballerinax/medium;
+
+configurable string openAIToken = ?;
+configurable string serviceUrl = ?;
+configurable string deploymentId = ?;
+configurable string mediumToken = ?;
+
+const API_VERSION = "2023-03-15-preview";
+
+public function main(string title) returns error? {
+    final medium:Client medium = check new ({auth: {token: mediumToken}});
+    final text:Client azureOpenAI = check new (
+        config = {httpVersion: http:HTTP_1_1, auth: {apiKey: openAIToken}},
+        serviceUrl = serviceUrl
+    );
+
+    medium:UserResponse response = check medium->getUserDetail();
+    medium:User? user = response.data;
+
+    if user !is medium:User {
+        return error("User is not a medium user");
+    }
+    string? userId = user.id;
+    if userId !is string {
+        return error("Medium user ID is not valid");
+    }
+
+    text:Deploymentid_completions_body completionsBody = {
+        prompt: string `Write a blog article on ${title} in markdown format.`,
+        max_tokens: 1000
+    };
+    text:Inline_response_200 completion = check azureOpenAI->/deployments/[deploymentId]/completions.post(API_VERSION, completionsBody);
+    string? content = completion.choices[0].text;
+
+    if content !is string {
+        return error("Generated content is not a string");
```

Comment:
Would `Failed to generate the content for the blog` or something similar would be a better error message?

Reply:
Addressed in [04771af](https://github.com/ballerina-guides/ai-samples/pull/62/commits/04771aff5856e3ca4265dba25968f3926268bd21)